spyder/plugins/ipythonconsole/widgets/client.py

```diff
@@ -232,6 +239,25 @@ def remove_stderr_file(self):
         except Exception:
             pass
 
+    @Slot()
+    def poll_stderr_file_change(self):
+        """Check if the stderr file just changed"""
+        try:
+            mtime = os.stat(self.stderr_file).st_mtime
+        except Exception:
+            return
+
+        if mtime == self.stderr_mtime:
+            return
+        self.stderr_mtime = mtime
+        try:
+            stderr = self._read_stderr()
+        except Exception:
+            stderr = None

```

Comment:
I added a method to do this in PR #12972. I'm going to merge that one first so you can rebase this one on top and take advantage of it.

Reply:
Please merge with the latest `4.x` to fix the conflict errors and also to use here the new method I added in PR #12972 called `get_stderr_contents`.